c-dependencies/js-compute-runtime/builtins/backend.cpp

```diff
@@ -0,0 +1,705 @@
+#include <iostream>
+#include <optional>
+
+// TODO: remove these once the warnings are fixed
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Winvalid-offsetof"
+#include "js/experimental/TypedData.h"
+#pragma clang diagnostic pop
+
+#include "backend.h"
+#include "host_call.h"
+#include "js/Conversions.h"
+
+bool isDigit(char character) { return character >= 48 && character <= 57; }
+bool isLowerAlpha(char character) { return character >= 97 && character <= 122; }
+bool isUpperAlpha(char character) { return character >= 65 && character <= 90; }
+bool isAlphaNumeric(char character) {
+  if (!isDigit(character) && !isLowerAlpha(character) && !isUpperAlpha(character)) {
+    return false;
+  }
+  return true;
+}
+bool isDash(char character) { return character == 45; }
+bool isDot(char character) { return character == 46; }
```

Comment:
What do you think about switching to the functions in the `<cchar>` header instead? (There are already definitions for things like `std::is_digit` in there.)

Reply:
oh yes this is nice - I've now moved to using `<cctype>` and `std::isalnum` instead of our own implementation -- [`bb1abe2` (#250)](https://github.com/fastly/js-compute-runtime/pull/250/commits/bb1abe28117aa02c48e73dd6d3f54c32488c7893)